File: python/blog/ds/queueLinkedList.py
from node import Node

class QueueLinkedList:
    head = None
    tail = None

    def toString(self):
        result = []
        tmp = self.head

        while(tmp is not None):
            result.append(str(tmp.data))
            tmp = tmp.next

        return ','.join(result)

    # addLast
    def enqueue(self, data):
        node = Node(data)
        
        if self.tail is not None:
            self.tail.next = node

        if self.head is None:
            self.head = node

        self.tail = node

    # removeFirst
    def dequeue(self):
        if self.empty():
            return None

        delete = self.head
        result = delete.data
        newHead = delete.next

        self.head = newHead

        if (newHead is None):
            self.tail = None

        delete = None
        return result


    def empty(self):
        return self.head is None

File: java/src/blog/ds/QueueLinkedList.java
package blog.ds;

import java.util.StringJoiner;

public class QueueLinkedList {
    Node head = null;

    Node tail = null;

    public String toString() {
        StringJoiner joiner = new StringJoiner(",");
        Node tmp = this.head;

        while (tmp != null) {
            joiner.add(Integer.toString(tmp.data));
            tmp = tmp.next;
        }

        return joiner.toString();
    }

    public void enqueue(int data) {
        // addLast
        Node node = new Node(data);

        if (this.tail != null)
            this.tail.next = node;

        if (this.head == null)
            this.head = node;

        this.tail = node;
    }

    public Integer dequeue() {
        // removeFirst
        if (this.empty())
            return null;

        Node delete = this.head;
        int result = delete.data;
        Node newHead = delete.next;

        this.head = newHead;

        if (newHead == null)
            this.tail = null;

        delete = null;
        return result;
    }

    public boolean empty() {
        return this.head == null;
    }
}
